src/coingecko/Coingecko.ts

```diff
@@ -150,6 +151,18 @@ export class Coingecko {
     assert(tokenPrice !== undefined);
     return [tokenPrice.timestamp.toString(), tokenPrice.price];
   }
+
+  async getCurrentPriceById(contractAddress: string, currency = "usd"): Promise<number> {
+    const coingeckoId = getCoingeckoTokenIdByAddress(contractAddress);
```

Comment:
Can/Should we do caching of prices here similar to [L127](https://github.com/across-protocol/sdk/blob/16e5906dbd4d58dc9d5e947bc28c0b0b8a3dc2a8/src/coingecko/Coingecko.ts#L127)?

Reply:
I think we should yeah

Reply:
We can/should - but I think this should be a follow-on. This function follows similar logic to the code below. My recommendation is that we should get this out and fast-follow after the AZERO launch. 

https://github.com/across-protocol/sdk/blob/16e5906dbd4d58dc9d5e947bc28c0b0b8a3dc2a8/src/coingecko/Coingecko.ts#L93-L116 and is

Reply:
aren't we going to make this request every single run though? Isn't that going to result in a lot of CG reqs?

Reply:
Ah shoot you're right this will be used in the relayer.